Mipa/Parser.cs

```diff
@@ -1,37 +1,175 @@
-﻿using System.Text;
+﻿using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace Mipa;
 
 public class MipaParser : IMipaParser
 {
     public ParseResult Parse(string content)
     {
-        if(string.IsNullOrWhiteSpace(content))
+        // Match all key value pairs
+        var memory = content.AsMemory();
+        var maxArguments = memory.Span.Count(':');
+
+        if (maxArguments == 0)
         {
             return new ParseResult
             {
-                Arguments = [],
+                Arguments = Array.Empty<KeyValuePair<ReadOnlyMemory<char>, ReadOnlyMemory<char>>>(),
                 Content = content,
             };
         }
 
-        // match all key value pairs
-        var matches = KeyValueRegex.Get().Matches(content);
-        var arguments = new List<KeyValuePair<string, string>>();
-        var builder = new StringBuilder(content);
-        for (var i = matches.Count - 1; i >= 0; i--)
-        {
-            var match = matches[i];
-            var key = match.Groups[1].Value;
-            var value = match.Groups[2].Value.Trim('"');
-            arguments.Add(new KeyValuePair<string, string>(key, value));
-            builder.Remove(match.Index, match.Length);
+        var arguments = new KeyValuePair<ReadOnlyMemory<char>, ReadOnlyMemory<char>>[maxArguments];
```

Comment:
**TODO:** Limit the maxArguments array size. Maybe change this back to a list so it auto-scales when the limit is reached.